### ISSUE TITLE
Forecast function for `EpiAware`

### ISSUE THREAD

**Issue by SamuelBrand1** [CDCgov/Rt-without-renewal#243]
## Current state of code

As per #239 we have a neat forecast function for models which have had inference on some time span $(1, T)$. The posterior chain can be used to condition the sampling of the model extended to a longer time span $(1, T + n)$. This is used in the analysis pipeline.

## Lack of generality

The problem here is that under the hood there are hard coded options, which reflect our analysis plan but are too specific to promote this function to `EpiAware`.

Pain points:

- `define_epiprob`: hard codes in discretised Gamma distributions.
- Lack of testing in a wider range of settings.

## Proposal

If we can fix these pain points then we can add this nice functionality to `EpiAware`.

**Comment by SamuelBrand1** [comment]
Worth noting that this forecast approach works _for our use case but is not general!_

https://github.com/TuringLang/Turing.jl/issues/2239#issuecomment-2144614352

**Comment by SamuelBrand1** [comment]
Does this 

> Worth noting that this forecast approach works _for our use case but is not general!_
> 
> [TuringLang/Turing.jl#2239 (comment)](https://github.com/TuringLang/Turing.jl/issues/2239#issuecomment-2144614352)

suggest that we move towards not having vectorised calls? e.g. no `MvNormal` calls?

**Comment by seabbs** [comment]
Yes. Do you know what implications this has (performance / limiting flexibility etc?)

**Comment by SamuelBrand1** [comment]
The main argument against would be AD performance; because known vector-form functions might have adjoints which are accessed rather than calculated.

**Comment by SamuelBrand1** [comment]
Flexibility would only be improved.
